base_layer/chat_ffi/src/contacts.rs

```diff
@@ -88,24 +90,30 @@ pub unsafe extern "C" fn check_online_status(
     receiver: *mut TariAddress,
     error_out: *mut c_int,
 ) -> c_uchar {
+    println!("chat_ffi: check_online_status ...");
     let mut error = 0;
     ptr::swap(error_out, &mut error as *mut c_int);
 
     if client.is_null() {
         error = LibChatError::from(InterfaceError::NullError("client".to_string())).code;
         ptr::swap(error_out, &mut error as *mut c_int);
+        return 0;
     }
 
     if receiver.is_null() {
         error = LibChatError::from(InterfaceError::NullError("receiver".to_string())).code;
         ptr::swap(error_out, &mut error as *mut c_int);
+        return 0;
     }
 
     let rec = (*receiver).clone();
     let result = (*client).runtime.block_on((*client).client.check_online_status(&rec));
 
     match result {
-        Ok(status) => status.as_u8(),
+        Ok(status) => {
+            println!("chat_ffi: check_online_status ({:?})", status);
+            status.as_u8()
+        },
```

Comment:
?

Reply:
Chasing an issue in the cucumber FFI

Reply:
I have been seeing this on CI for quite some time:
```
process didn't exit successfully: `/runner/_work/tari/tari/target/release/deps/cucumber-6dca1820c03c28fb 
-t '(@wallet-ffi or @chat-ffi) and @critical and (not @long-running) and (not @broken)' -c 1 --retry 2` 
(signal: 11, SIGSEGV: invalid memory reference)
```
I believe it may be the same issue I am seeing on my system:
```
process didn't exit successfully: `C:\Users\pluto\.tari\Code\@tari-project\target\release\deps\cucumber-c5ef6df7eed7d10e.exe 
--tags @chat-ffi` (exit code: 0xc0000005, STATUS_ACCESS_VIOLATION)
```